### PR TITLE
[CF-433] Handle image deleted from glance backend.

### PR DESCRIPTION
**Pull request by antonf**
Glance use some sort of backend for storing images. It can be
Swift/Ceph/filesystem. Image file can be deleted from this backend
through SSH for filesystem backend, or through Swift API for Swift
backend or through Librados for Ceph backend. Glance will not that
image was deleted and will consider the image as active, but will
raise 404 error when you will try to download image file.
This patch handle such situation, falling back to restoring image
from compute nodes.

### CODE REVIEW

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check failed
http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry-unittests-and-style-checks/3308/

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferry.lib.os.discovery.nova
cloudferry/lib/os/discovery/nova.py:198: [W0612(unused-variable), _list_ephemeral] 
Unused variable 'format'

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by plomakin** [comment]
retest this please

**Comment by plomakin** [comment]
Tested migrate2 action on devlab - it's working correctly

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

File: cloudferry/lib/utils/local_db.py [hunk 1]
```diff
@@ -79,7 +79,8 @@ def _initialize(self):
             self._tls.top_level = self
             self._conn = sqlite3.connect(
                 SQLITE3_DATABASE_FILE,
-                detect_types=sqlite3.PARSE_DECLTYPES | sqlite3.PARSE_COLNAMES)
+                detect_types=sqlite3.PARSE_DECLTYPES | sqlite3.PARSE_COLNAMES,
+                timeout=3600)
```
Comment (SVilgelm) [inline cloudferry/lib/utils/local_db.py]: Do we really need 1 hour timeout?

**Comment by SVilgelm** [comment]
👍 But add a description for the commit and pull request.

**Comment by MirantisWorkloadMobility** [comment]
$PYLINT_STATUS

**Comment by SVilgelm** [comment]
Thanks

**Pull request merged**